runtime/src/rpc.rs

```diff
@@ -1747,19 +1600,15 @@ impl VaultRegistryPallet for InterBtcParachain {
     /// * `uri` - URI to the client release binary
     /// * `checksum` - The SHA256 checksum of the client binary
     async fn set_current_client_release(&self, uri: &[u8], checksum: &H256) -> Result<(), Error> {
-        self.with_unique_signer(|signer| async move {
-            let release = ClientRelease {
-                uri: uri.to_vec(),
-                checksum: *checksum,
-            };
-
-            self.api
-                .tx()
+        let release = ClientRelease {
+            uri: uri.to_vec(),
+            checksum: *checksum,
+        };
+        self.with_unique_signer(
+            metadata::tx()
                 .clients_info()
-                .set_current_client_release(uri.to_vec(), release)
-                .sign_and_submit_then_watch_default(&signer)
-                .await
-        })
+                .set_current_client_release(uri.to_vec(), release),
```

Comment:
not related to your pr but I think this is wrong - this should be e.g. "vault" Can you add a comment maybe?